src/protocol.rs

```diff
@@ -52,6 +52,8 @@ const WS_WITH_PATH: u32 = 4770; // Note: not standard
 const WSS: u32 = 478;
 const WSS_WITH_PATH: u32 = 4780; // Note: not standard
 
+type Multihash = MultihashGeneric<64>;
```

Comment:
Can you document where the `64` is coming from?

Reply:
I don't know why it is 64. That was the value set on the `Code` enum so I just retained that.

Reply:
`MultihashGeneric` is generic over the allocation size of the digest.

Now what that means in practice and whether this is architecture dependent, I don't know.

I don't think we should hard code this magic number here without knowing what it actually means.

Maybe @vmx can help us here?

Reply:
It depends on the hashes you use. It's the maximum size a hash can have. So e.g. if all your hashes are at most 256-bit you can use 32.

Reply:
> > I don't think we should hard code this magic number here without knowing what it actually means.
> 
> Perhaps there is a misunderstanding here. The multihash we used before was also hard-coded to 64 because `mutlihash::Code` specified `64` in the custom-derive: https://github.com/multiformats/rust-multihash/blob/master/src/multihash_impl.rs#L12
> 
> The code here is equivalent to what it was before.

No misunderstanding. The issue I see is, that in case `64` ever changes (that is `type Multihash = MultihashGeneric<64>;`) for whatever reason within `multihash`, we would probably miss updating our value here.

Is that a contrived example? Will we never use a hasher with a size beyond 64?

Is there any way we can ensure to stay consistent? E.g. through a unit test across the size of the hashes we use?

Reply:
> > > I don't think we should hard code this magic number here without knowing what it actually means.
> > 
> > Perhaps there is a misunderstanding here. The multihash we used before was also hard-coded to 64 because `mutlihash::Code` specified `64` in the custom-derive: https://github.com/multiformats/rust-multihash/blob/master/src/multihash_impl.rs#L12
> > 
> > The code here is equivalent to what it was before.
> 
> No misunderstanding. The issue I see is, that in case `64` ever changes (that is `type Multihash = MultihashGeneric<64>;`) for whatever reason within `multihash`, we would probably miss updating our value here.

It doesn't matter what the `multihash` library does, I think? Once the split of `multihash` is complete, it won't specify a size any more and just expose the generic data structure.

> Is that a contrived example? Will we never use a hasher with a size beyond 64?

Eventually, the only use of `Multihash` in here will be the `/certhash` protocol because `/p2p` will use `PeerId`.

The webrtc and webtransport specs only allow SHA256 at the moment, right? So we'd be fine with 32 size actually. If we ever update those specs to allow for a hash that requires more than 64 bytes, we just need to release a breaking change of this lib.

That is the only implication here, I don't think we need to be consistent across the ecosystem in all usages of `Multihash`, otherwise it wouldn't need to be generic in the first place.

Reply:
The reasoning above makes sense to me. Thanks @thomaseizinger. Would you mind adding the above as a comment to the type alias for future us? That would be enough for me to move forward here.

Reply:
Let me know what you think of a55f8ac.